### ISSUE TITLE
SegmentedControl does not work with an empty array

### ISSUE THREAD

**Issue by hckhanh** [mantinedev/mantine#2400]
### What package has an issue

@mantine/core

### Describe the bug

When I use segment with an initial empty array, I get this bug:

```jsx
<SegmentedControl data={[]} value={'any value'} />
```

```
SegmentedControl.tsx:99 Uncaught TypeError: Cannot read properties of undefined (reading 'value')
    at SegmentedControl.tsx:99:140
    at renderWithHooks (react-dom.development.js:16305:18)
    at updateForwardRef (react-dom.development.js:19226:20)
    at beginWork (react-dom.development.js:21636:16)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at HTMLUnknownElement.__trace__ (inline-script-content.js:146:30)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at beginWork$1 (react-dom.development.js:27451:7)
    at performUnitOfWork (react-dom.development.js:26557:12)
```

### What version of @mantine/hooks page do you have in package.json?

5.2.5

### Do you know how to fix the issue

No

### Possible fix

I think it should allow the empty input data.

**Comment by hckhanh** [comment]
Another bug is If i use empty string `''` as value, it does not work at all

```jsx
import { SegmentedControl } from '@mantine/core';

function Demo() {
  return (
    <SegmentedControl
      data={[
        { label: 'All', value: '' },
        { label: 'React', value: 'react' },
        { label: 'Angular', value: 'ng' },
        { label: 'Vue', value: 'vue' },
        { label: 'Svelte', value: 'svelte' },
      ]}
    />
  );
}
```

**Comment by rtivital** [comment]
Fixed in 5.3.1
